### ISSUE TITLE
Delta index jobs cause `rake ts:index` to fail.

### ISSUE THREAD

**Issue by bryantraywick** [pat/ts-sidekiq-delta#4]
Processing delta jobs while a full index is running leads to "Resource temporarily unavailable" errors with the main indexer. The problem is that the temporary delta index spl file is already locked by the delta index jobs and the full index exits.

I can recreate this issue 100% of the time by calling save on an ActiveRecord object in a loop and then running `rake ts:index`.

I think the guard file feature in thinking-sphinx isn't catching this scenario properly and [this commit](https://github.com/pat/thinking-sphinx/commit/f65106cedd241af7edafbc4f63d637bf20b115c9) in particular may be at fault. When running a full index the guard file that gets created is `ts---all.tmp`, but the delta index jobs only check for the specific index guard file. With the current guard file implementation there is a also a chance of [a race condition](https://github.com/pat/thinking-sphinx/blob/master/lib/thinking_sphinx/controller.rb#L6-L9) where the check for the guard file is performed and then the guard file is created. After the guard file check and before the creation, a delta job could create the guard file that didn't previously exist. This can lead to multiple delta jobs attempting to index at the same time, or a delta job and `ts:index` updating the same index.

Putting aside the race condition concerns, I think a better approach would be to create guard files for each index as they are being processed. This would likely require multiple calls out to the Sphinx indexer command, one for each index core and delta, but would allow for delta updates to other indexes to the processed during a long running full index.

Versions:

Sphinx: 2.1.9
thinking-sphinx: 3.1.1
ts-sidekiq-delta: 7a9bac171c8ae11d38179104b1a60589bd047e1d

**Comment by akshay-vishnoi** [comment]
hey, is there any update on this issue?  I am also facing the same issue.

**Comment by pat** [comment]
It's taken a long while, but finally I have a fix in place for this: pat/thinking-sphinx@d12e74130a131e9b8f2ac60a250182da5d13215e

If you update to the latest in the `develop` branch of the Thinking Sphinx repo, you'll have that commit available to you, and then you can add the following to an initializer:

```ruby
ThinkingSphinx::Configuration.instance.indexing_strategy = ThinkingSphinx::IndexingStrategies::OneAtATime
```
